arrow-fx-reactor/src/main/kotlin/arrow/fx/reactor/MonoK.kt

```diff
@@ -225,31 +264,29 @@ data class MonoK<out A>(val mono: Mono<out A>) : MonoKOf<A> {
      *
      * fun main(args: Array<String>) {
      *   //sampleStart
-     *   val result = SingleK.cancellable { cb: (Either<Throwable, String>) -> Unit ->
+     *   val result = MonoK.cancellable { cb: (Either<Throwable, String>) -> Unit ->
      *     val nw = NetworkApi()
      *     val disposable = nw.async { result -> cb(Right(result)) }
-     *     SingleK { disposable.invoke() }
+     *     MonoK { disposable.invoke() }
      *   }
      *   //sampleEnd
      *   result.value().subscribe(::println, ::println)
      * }
      * ```
      */
     fun <A> cancellable(fa: ((Either<Throwable, A>) -> Unit) -> CancelToken<ForMonoK>): MonoK<A> =
-      Mono.create { sink: MonoSink<A> ->
-        val cb = { either: Either<Throwable, A> ->
-          either.fold(sink::error, sink::success)
-        }
-
-        val token = try {
-          fa(cb)
-        } catch (t: Throwable) {
-          cb(Left(t.nonFatalOrThrow()))
-          just(Unit)
+      Flux.create<A> { sink ->
```

Comment:
Hehe. I was unable to implement this operator using `Mono` :/

It works fine through `Flux`. I'd opt to create a separate ticket for this, since it's an optimization rather than incorrect behavior.